conveyor/worker.go

```diff
@@ -0,0 +1,525 @@
+	// TODO: how to do it?
+	// catch conveyor lock, check input queue, if It's empty - remove slot from map, if it's not - got to Working state
+}
+
+func (w *worker) processSignalsSuspending(elements []queue.OutputElement) int {
+	log.Debugf("[ processSignalsSuspending ] starts ... ( pulseState: %s )", w.slot.pulseState.String())
+	numSignals := 0
+	// TODO: add check if many signals come
+	for i := 0; i < len(elements); i++ {
+		el := elements[i]
+		if el.IsSignal() {
+			numSignals++
+			switch el.GetItemType() {
+			case PendingPulseSignal:
+				log.Warn("[ processSignalsSuspending ] Must not be PendingPulseSignal here. Skip it")
+			case ActivatePulseSignal:
+				w.changePulseState()
+				w.slot.slotState = Initializing
+				w.activatePulseSync = el.GetData().(queue.SyncDone)
+				log.Info("[ processSignalsSuspending ] Set slot state to 'Initializing'")
+			case CancelSignal:
+				w.stop = true // TODO: do it more correctly
+			default:
+				panic(fmt.Sprintf("[ processSignalsSuspending ] Unknown signal: %+v", el.GetItemType()))
+			}
+		} else {
+			break
+		}
+	}
+
+	return numSignals
+}
+
+func (w *worker) readInputQueueSuspending() error {
+	log.Debugf("[ readInputQueueSuspending ] starts ... ( pulseState: %s )", w.slot.pulseState.String())
+	elements := w.slot.inputQueue.RemoveAll()
+	numSignals := w.processSignalsSuspending(elements)
+
+	// remove signals
+	elements = elements[numSignals:]
+
+	for i := 0; i < len(elements); i++ {
+		el := elements[i]
+
+		_, err := w.slot.createElement(GetStateMachineByType(InputEvent), 0, el)
+		if err != nil {
+			return errors.Wrap(err, "[ readInputQueueSuspending ] Can't createElement")
+		}
+	}
+
+	if len(elements) != 0 && w.slot.pulseState == constant.Past {
+		w.slot.slotState = Working
+		log.Info("[ readInputQueueSuspending ] Set slot state to 'Working'")
+	}
+
+	return nil
+}
+
+func (w *worker) suspending() {
+	log.Debugf("[ suspending ] starts ... ( pulseState: %s )", w.slot.pulseState.String())
+	switch w.slot.pulseState {
+	case constant.Past:
+		w.sendRemovalSignalToConveyor()
+	case constant.Present:
+		w.calculateNodeState()
+	}
+	for w.slot.isSuspending() {
+		err := w.readInputQueueSuspending()
+		if err != nil {
+			panic(fmt.Sprintf("[ suspending ] readInputQueueSuspending. Can't readInputQueueSuspending: %s", err))
+		}
+	}
+
+	log.Infof("[ suspending ] Leaving suspending. pulseState: %s. slotState: %s",
+		w.slot.pulseState.String(),
+		w.slot.slotState.String(),
+	)
+}
+
+func (w *worker) migrate(status ActivationStatus) error {
+	log.Infof("[ migrate ] Starts ... ( status: %s. pulseState: %s )", status.String(), w.slot.pulseState.String())
+	numElements := w.slot.len(status)
+	for ; numElements > 0; numElements-- {
+		element := w.slot.popElement(status)
+		migHandler := element.stateMachineType.GetMigrationHandler(w.slot.pulseState, element.state)
+		payLoad, newState := element.payload, element.state
+		var err error
+		if migHandler == nil {
```

Comment:
In this condition payLoad and newState doesn't change, so you can just do all logic after condition (setNewElementState and pushElement). It more clear than continue and if/else. (you can just check `if migHandler != nil` and do call, without `else` branch)